src/controllers/controller.doughnut.js

```diff
@@ -138,6 +138,37 @@ module.exports = DatasetController.extend({
 		return ringIndex;
 	},
 
+	// Get radius length offset of the dataset in relation to the visible datasets weights. This allows determining the inner and outer radius correctly
+	getRingWeightedOffset: function(datasetIndex) {
```

Comment:
I would recommend making all these new methods private. I.e. prefix the name with `_` and mark as private in the jsdoc:
```
/**
 * @private
 */
```

See `_resolveElementOptions` in this file as an example.